awscli/customizations/cloudfront.py

```diff
@@ -33,6 +34,41 @@ def register(event_handler):
         'operation-args-parsed.cloudfront.create-invalidation',
         validate_mutually_exclusive_handler(['invalidation_batch'], ['paths']))
 
+    update_distribution_args = [
+        ExclusiveArgument(
+            'default-root-object', 'distribution-config', help_text=(
+                'The object that you want CloudFront to return (for example, '
+                'index.html) when a viewer request points to your root URL.')),
+        ]
+    _setup_custom_params(
+        event_handler, 'update-distribution', update_distribution_args,
+        _update_distribution)
+
+    create_distribution_args = update_distribution_args + [
+        ExclusiveArgument(
+            'origin-domain-name', 'distribution-config',
+            help_text='The domain name for your origin.'),
+        ]
+    _setup_custom_params(
+        event_handler, 'create-distribution', create_distribution_args,
+        _create_distribution)
+
+
+def _setup_custom_params(event_handler, subcmd, custom_args, callback):
+    # Usage: _setup_custom_params(..., "create-distribution", [...], callable)
+    event_handler.register(
+        'operation-args-parsed.cloudfront.%s' % subcmd,
+        validate_mutually_exclusive_handler(
+            [arg.py_name for arg in custom_args], ['distribution_config']))
+    event_handler.register(
+        'building-argument-table.cloudfront.%s' % subcmd,
+        lambda **kwargs: _add_exclusive_arguments(args=custom_args, **kwargs))
+    event_handler.register('calling-command.cloudfront.%s' % subcmd, callback)
```

Comment:
The typical way this is done with other customizations is to override `add_to_params` in a custom arg class.  Is there a reason this needs to be different from the other customizations and use `calling-command`?  Given you already have a custom arg class `ExclusiveArgument`, it doesn't seem like you gain anything by using `calling-command`.